### ISSUE TITLE
feat: Add CITL export for OIP integration

### ISSUE THREAD

**Issue by noahgift** [paiml/bashrs#83]
## Summary

Add Compiler-In-The-Loop (CITL) export support to enable integration with organizational-intelligence-plugin (OIP) for defect pattern analysis and training data generation.

## Background

OIP uses compiler diagnostics as ground-truth labels for ML-based defect classification. Depyler already implements CITL export. This ticket adds the same capability to rash (Bash→Rust transpiler).

### CLI Flag
```bash
rash transpile script.sh --citl-export diagnostics.json
```

### Export Format
```rust
#[derive(Serialize)]
pub struct CitlExport {
    pub source_file: String,
    pub error_code: Option<String>,      // e.g., "E0308"
    pub clippy_lint: Option<String>,     // e.g., "clippy::unwrap_used"
    pub level: String,                   // "error" | "warning"
    pub message: String,
    pub oip_category: Option<String>,    // Pre-classified if known
    pub confidence: f32,
    pub span: Option<SpanInfo>,
    pub suggestion: Option<SuggestionInfo>,
    pub timestamp: i64,
}
```

### Bash-Specific Considerations
- Map shell command errors to structured diagnostics
- Capture subshell/pipeline conversion failures
- Track variable expansion → Rust string handling issues
- Process substitution edge cases

## Effort
~2-4 hours

## Related
- OIP CITL spec: `organizational-intelligence-plugin/src/citl.rs`
- Depyler implementation: `depyler/crates/depyler-core/src/citl.rs`
